x/pairing/keeper/delegator_rewards_test.go

```diff
@@ -0,0 +1,238 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/testutil/common"
+	"github.com/lavanet/lava/utils/sigs"
+	"github.com/lavanet/lava/utils/slices"
+	dualstakingtypes "github.com/lavanet/lava/x/dualstaking/types"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	"github.com/lavanet/lava/x/pairing/types"
+	"github.com/stretchr/testify/require"
+)
+
+// TestProviderDelegatorsRewards tests that the provider's reward (considering delegations) is as expected
+// Also, it checks that the delegator reward map is updated as expected
+func TestProviderDelegatorsRewards(t *testing.T) {
```

Comment:
this needs to be repeated with multiple combinations:
(would cover the test explicit test cases in the next two functions)

commission = 0, 47, 100  (the "47" chosen to be unlikely to nicely divide the effective delegation)
limit = 0, larger than total delegation, smaller than total delegation
delegation ratios (between two delegators):   0/100, 10/90, 50/50  (the 14/86 is unlikely to nicely divide the total delegations)

so that's a matrix of 3x3x3 combinations to test. 
it will check edge cases, including the case of unclaimed leftovers that need to go to the provider.

Reply:
done. see https://github.com/lavanet/lava/commit/57489eca55ecc1b76b193a712bd87bb88e84f30b

Reply:
nice. (I was thinking of a 3-deep nested loops, of 3 cases each, but this works just as well.)